Amplify/Categories/DataStore/Model/Embedded.swift

```diff
@@ -0,0 +1,28 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+// MARK: - CustomCodable
+
+/// All persistent custom types should conform to the CustomCodable protocol.
+public protocol Embedded: Codable {
```

Comment:
minor: maybe worth considering calling the protocol `Embeddable`.

It suits well what the industry usually uses (see [Hibernate approach](https://www.baeldung.com/jpa-embedded-embeddable)) and also the iOS naming patterns.

An `Embeddable` is `.embedded(type:)` in a `Model`. I can see that making way more sense than "custom type" or "non-model type". =)

Reply:
that sounds good, i will make this change!